copy/validate_gzip_test.go

```diff
@@ -0,0 +1,57 @@
+package copy
+
+import (
+	"compress/gzip"
+	"crypto/sha256"
+	"encoding/hex"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Verify that the GZIP writer results are consistent across golang versions
+//
+// * a large payload is used to ensure the compression code utilizes several blocks
+// * a small payload forces flushes in the compression code by using a partial block
+//
+// Verified against: go 1.7.3, go 1.6.3
+func TestGzipWriter(t *testing.T) {
+	cases := []struct {
+		name   string
+		digest string
+	}{
+		{"fixtures/random_512k.bin", "12370c316d20ce7c37ef9f3ffd8cc5a08334b22abd2ae922759ed9511fad245d"},
+		{"fixtures/random_500b.bin", "14e61c568333905163dcc8bacefebb4d4d2ca750fe7d5d7eb64c8ed4f4ca4b85"},
+	}
+
+	for _, test := range cases {
+		payload, err := ioutil.ReadFile(test.name)
+
+		output, err := ioutil.TempFile(os.TempDir(), "gzipWriterTest_")
```

Comment:
Couldn’t this just use a `bytes.Buffer` or something? (Not blocking as such, and does not resolve the fundamental issues of course.)